app/views/devise/registrations/new.html.erb

```diff
@@ -0,0 +1,81 @@
+<div class="flex items-center justify-center h-screen px-6 bg-gray-200 mt-5">
+  <div class="w-full max-w-sm p-6 bg-white rounded-md shadow-md">
+    <div class="flex items-center justify-center">
+      <%= svg("flame", class: "w-10 h-10") %>
+
+      <span class="text-3xl font-semibold text-gray-700">AgroReports</span>
+    </div>
+    
+    <h2 class="text-2xl font-semibold text-gray-700">Sign up</h2>
+    <%= form_for(resource.with_account, as: resource_name, url: registration_path(resource_name), class: "mt-4") do |f| %>
+      <div class="block">
+        <%= f.label :name, class: "text-sm text-gray-700" %>
+        <%= f.text_field :name, autofocus: true, class: "block w-full mt-1 border-gray-200 rounded-md focus:border-indigo-600 focus:ring focus:ring-opacity-40 focus:ring-indigo-500" %>
+        <% resource.errors.full_messages_for(:name).each do |message| %>
+          <span class="font-light text-sm text-red-500">
+            <%= message %>
+          </span>
+        <% end %>
+      </div>
+
+      <div class="block mt-3">
+        <%= f.label :email, class: "text-sm text-gray-700" %>
+        <%= f.email_field :email, autocomplete: "email", class: "block w-full mt-1 border-gray-200 rounded-md focus:border-indigo-600 focus:ring focus:ring-opacity-40 focus:ring-indigo-500" %>
+        <% resource.errors.full_messages_for(:email).each do |message| %>
+          <span class="font-light text-sm text-red-500">
+            <%= message %>
+          </span>
+        <% end %>
+      </div>
+
+      <div class="block mt-3">
+        <%= f.label :password, class: "text-sm text-gray-700" %>
+        <% if @minimum_password_length %>
+          <em class="text-sm text-gray-700">
+            (<%= @minimum_password_length %> characters minimum
+          </em>
+        <% end %>
+        <%= f.password_field :password, autocomplete: "new-password", class: "block w-full mt-1 border-gray-200 rounded-md focus:border-indigo-600 focus:ring focus:ring-opacity-40 focus:ring-indigo-500"%>
+        <% resource.errors.full_messages_for(:password).each do |message| %>
+          <span class="font-light text-sm text-red-500">
+            <%= message %>
+          </span>
+        <% end %>
+      </div>
+
+      <div class="block mt-3">
+        <%= f.label :password_confirmation, class: "text-sm text-gray-700" %>
+        <%= f.password_field :password_confirmation, autocomplete: "new-password", class: "block w-full mt-1 border-gray-200 rounded-md focus:border-indigo-600 focus:ring focus:ring-opacity-40 focus:ring-indigo-500" %>
+        <% resource.errors.full_messages_for(:password_confirmation).each do |message| %>
+          <span class="font-light text-sm text-red-500">
+            <%= message %>
+          </span>
+        <% end %>
+      </div>
+
+      <div class="block mt-3">
+        <%= f.fields_for :account do |account| %>
```

Comment:
this is not a request for change, just a comment.

Instead of the `resource.with_account` you did above, you  could also have initialized the account in here, with:

`f.fields_for :account, Account.new(owner: resource)...`